### DISCUSSION TITLE
Why is open-Shell not developed well enough

### DISCUSSION THREAD

**Discussion by ullireinders** [Open-Shell/Open-Shell-Menu#1137] [Q&A] [answered]
Hi there!
Today I will renew my request, which was deleted from Coddec.

Yesterday I thought I could update Open-Shell and write an article for my blog. But I gave up on that when I saw that this program is still not ready to be configured really well.

Why has OS 35 languages, when all things will be displayed in english ?
Why is it so difficult/not possible to ask you personally questions about this program ?
Why has OS after all these version not a larger collection of start-buttons to select ?
It is much to complicated to set an other start-Button. They will be displayed not correct.
I have Windows 10 21H2 and OS 4.4.170.

I would be very happy about answers to my questions. KR, Ulli

**Comment by CTVCAM8** [comment]
About more internal start buttons to select from.

Not a coder. Just my way of looking at it.
It would be great if there could be a few more buttons to choose from, 
maybe even all code similar to the Windows 11 start button. 

Looking inside the start menu files though, 
I see at least 4 different sized shell start buttons of the same design by Ivo, 
which covers most display sizes (to avoid blurry images or incorrect size matches).
  
So, if a graphic designer donates their work to Open-Shell, 
along with coders adding the extra start-button option to select, 
the designer would be required to make 4 sizes too, 
as well as not infringing on any copyrights.
A few extra start buttons can quickly add up to 12 extra images!


For myself, I don't find it that difficult to set up an external start button.
About the only annoyance would be adjusting the "Button size" option 
when the image does not match up to the display size.
One has to try a fair number of times to get it right. 

I don't know.
Maybe there could be more info put in the Help file for "Custom Start Button"
to include templates with pixel dimensions for various display sizes.

**Comment by CTVCAM8** [comment]
Ok, that's weird. I got an email from @Blackcrack and clicked "View it on GitHub" 
but @Blackcrack comments didn't show up yet. 
Here are comments and a picture from @Blackcrack: 

----------------------------------------------------

[the size must not in 3 diffrent Pictures, 
the magic word is scaling and set a size to easy scaling via code and setting up the size of the startbutton, 
herewith it is given to set up a spezially size and this could happens with a slider
and a "liveview" by side on the sraermenu buttons, how it changing it .. by sliding the slider, 
therewith with a extra-slider for height adjustment for adjustment of the size of the button, 
more downward and more upward for more elongated buttons like the Win96 Startbutton
or a button who stays more over the bar .... therewith would be give for the user the full control over the size and high of the button .]


![Blackcrack](https://user-images.githubusercontent.com/44127216/190047253-f4d7da3e-1157-4725-9db2-f42d5e2bd8f1.png)

-------------------------------------------------------------------------------------------------------------------------------------------

@CTVCAM8

I wasn't really talking about the old Classic start button that has normal and pressed actions,
where Windows makes the button and text size and possibly icon image that can vary in size with code. 
That's already in Open-Shell.
I'm sure some of that code (padding for sizing, text size, font, image, or icon image) could be made available for custom user control.


About scaling.
To scale correctly, there either has to be a large amount of exact display-sized images to choose from (Win7 has at least 10 various-sized start buttons),
or one very large-sized image to resize down. One size does not fit all in my opinion, as blurriness or pixelization will tend to show up.
For most people, maybe they don't notice. 

Yes.
A slider and real-time preview would be nice to adjust an internal or custom external start button no matter if it's a single frame
(one frame image is possible with a coded color pixel line at the top of an image), or the usual 3-frame animation button,
or up to 255 and beyond frames animation button with coded color pixel line at the top.

**Reply by blackcrack** [reply]
hmmm... @CTVCAM8 you could have corrected it.. \*bg* yes, i have delete it.. because, i would disturb, because Ulli had questions .. and should answered .. what is also important.. was easy a burb of ideas.. because i had delete it again.. how ever.. you have repost it again..
how ever, the Ask stand on top..

Scaling:
scaling, so, why not enable pictures with 256px.. i work with some pictures in Linux and SVG where i insert png's into svg and then scale it to smaller for Konqueror/Dolphin and Plasma as Icontheme

Quality of the Startbutton: 
thinking on the Peoples who have Glass on the nose ;) The quality of the Pictures, more bigger the three in long pictures it is more HD and UHD Quality is it, if it is make shrinking to the right size, therewith or because this should it make a possible for make it possible for take a more big size of a Start-menu Picture with 3 Pictures inside and shrink it to the right size then have we the possible for a HD/UHD Quality, if i think on to take a TV as Monitor.. like me, i have a 43" LG as Monitor.. but there be more huge TV's like 60" or 100" there should be the Start menu have a UHD-Quality and maybe a size in 300px or 400px and then shrink in the right scale for desktop to have a "UHD" quality as Start-Button and then is there not a size limitation of the Picture for Startmenu, because it is ever shrink in the right size in actual desktop-scale ...  
only the height and wide with 3 pictures in %, so 100% and 3,33% in wide, but Standard is 
one button 54px x 54px and the whole picture 162px high ... 
PNG, transparent (here showing as white)
( Santaclause Startbutton as Example)
![start_button](https://user-images.githubusercontent.com/7925717/190063637-9ba33343-edd1-4a71-885d-f6dae18ab0f1.png)
i have work here now with transparency, some peoples works with black background
and it is work with an png instead a Microsoft bmp , so it is also possible to use other formats ..
(because other ever speaks it must take the "standart" of MS.. and taking different specials... nope, must not be.. imho)
best

**Comment by CTVCAM8** [comment]
Sorry. I waited for hours.
Next time I won't respond until I see it on GitHub.

I wonder what method Open-Shell uses to resize the raster png/bitmap start button image? It's probably Nearest neighbor, definitely not Bicubic (sharper), 
and most definitely can't use a vector image. One person (Davie Stewart) on Classic Shell told me that a vector image resized to a smaller size is not perfect either.
They look woolly. :)

**Reply by blackcrack** [reply]
naa, it is the kind/type of the shrinking, there is different possibility's to shrink.. 
take the gimp, open a big picture and then shrink the picture, there is a pulldown menu for the kind/type of shrinking,
now be comes 2 more news of possibility of the shrinking added, more lossless .. and less woolly ..... who wo say, it works not, have only surfacely informations and blabla only something to seems "professional" (...pffhhhh :\ \*thumpsdown*) 


@CTVCAM8 because waiting and respond.. was okey with you reacting.. trust you feeling ! You have not to wait.. or you have not directly to answer.... you have not the duty... open source and the acting is more like a feeling.. a want do if want.. there is not a duty !.. so, if you want make an answer, can you do.. but is not a must.. we both be long enough at the net.. ;) we have enough fulfilled THE Duty to "make the Internet great again.." \*giggle* via modems :)
best

**Comment by ullireinders** [comment]
Can anyone tell me, why codded does not answer to this thread ?

**Reply by Ibuprophen** [reply]
Are you referring to @coddec?

**Reply by blackcrack** [reply]
Hi Ulli :) 
wenn du ein @ vor den Namen setzt, dann werden die leute direct angemailt
wie ich hier mache @ullireinders das Du hier fett siehst.. und bedeutet, 
daß es direct refferiert wird, soll heisen, bescheid gegeben an die jenige person mit Username..
was CTVCAM8 für Dich nun gemacht hat...
viele Grüße
best

**Reply by ullireinders** [reply]
> Hi Ulli :) wenn du ein @ vor den Namen setzt, dann werden die leute direct angemailt wie ich hier mache @ullireinders das Du hier fett siehst.. und bedeutet, daß es direct refferiert wird, soll heisen, bescheid gegeben an die jenige person mit Username.. was CTVCAM8 für Dich nun gemacht hat... viele Grüße best

Danke Dir für die Erklärung. Dann werde ich das jetzt malmachen, da er ja immer noch nichts dazu geschrieben hat!
LG, Ulli

**Comment by ullireinders** [comment]
Hello @coddec ! Could you please comment on the subject and answer my questions?
Thx so much, Ulli

**Comment by coddec** [comment, marked as answer]
>Why has OS 35 languages, when all things will be displayed in english ?

Better ask the original author Ivo, this is how it works when it was created in the first place.

>Why is it so difficult/not possible to ask you personally questions about this program ?

It is not that hard. Just create a GitHub account and ask here in discussion, if you are talking about the time to get a reply, all maintainers and other contributors here are volunteers, all got things to do and more importantly make living.  

>Why has OS after all these version not a larger collection of start-buttons to select ?

Anyone can contribute to the project. For questions like "Why this feature that feature is not impanated" please refer to the last question above

>It is much to complicated to set an other start-Button. They will be displayed not correct.

?
If you think there is bug in the software please submit a issue with screen shot or error message etc. and ways to reproduce the issue or problem, by simply saying "They will be displayed not correct" does not help to troubleshoot the issue, 

Thanks

**Reply by ullireinders** [reply]
Okay, my dear @coddec, then I would like to ask you what your prospects are for the near future of Open Shell. Is there a roadmap, a so-called timeline for Open-Shell ? What will come next, for example ? Or are you still correcting minor bugs and will only tackle bigger things after the bug fixes ? Tell us please!

**Reply by coddec** [reply]
It's a community driven software/project, everyone is welcome to contribute.
No specific roadmap or timeline at the moment.
If you want a feature or fix an issue, feel free to modify the code and contribute, submit the pull request etc.
Thanks

**Reply by ullireinders** [reply]
> It's a community driven software/project, everyone is welcome to contribute. No specific roadmap or timeline at the moment. If you want a feature or fix an issue, feel free to modify the code and contribute, submit the pull request etc. Thanks

My main focus at the moment would be for this tool to finally actively use the language catalogues available for it and not always show us everything only in English. How is this feasible ?

**Reply by coddec** [reply]
Please submit a pull request if you know how to make it happen in the code or create an issue and mark it as feature request.
Thanks

**Reply by ullireinders** [reply]
> Please submit a pull request if you know how to make it happen in the code or create an issue and mark it as feature request. Thanks

Unfortunately, it is not at all clear to me how I could do that. I am not a programmer! I am mainly interested in being able to better understand the configuration environement of the tool. This is primarily based on the fact that my English skills are not sufficient. In other words: I am not gifted in languages. If I have to write anything in English, I always need a translator (like deepl.com). Sorry!
